### PR TITLE
DOTCOM-3584: convert resource files and tests to typescript

### PR DESCRIPTION
**Pull request by lucasassisrosa**
I have labeled some files as `important` for check when reviewing. Please use the file filter to omit the "excluded files" when reviewing

- [ ] resources
- [ ] tests
- [x] Jest base setup
- [x] Error

## Manual testing

- run `npm run lint`
- run `npm run build`
- check types in `types` folder
- check output in `dist` folder
- check README instructions on testing and typescript
- run `npm run test`
- see test report

### CODE REVIEW

**Review by jonathan-telnyx — APPROVED** [review]

File: .github/workflows/node.js.yml
Comment (lucasassisrosa) [inline .github/workflows/node.js.yml:1]: important

File: .github/workflows/npm-publish.yml
Comment (lucasassisrosa) [inline .github/workflows/npm-publish.yml:1]: important

File: CHANGELOG.md
Comment (lucasassisrosa) [inline CHANGELOG.md:1]: important

File: README.md
Comment (lucasassisrosa) [inline README.md:1]: important

File: jest.config.js
Comment (lucasassisrosa) [inline jest.config.js:1]: important

File: package.json
Comment (lucasassisrosa) [inline package.json:1]: important

File: tsconfig.json
Comment (lucasassisrosa) [inline tsconfig.json:1]: important

**Review by DeividVeloso — APPROVED** [review]

**Review by ADandyGuyInSpace — APPROVED** [review]
We have a taskforce currently to double check the openapi specs against the live and correct them, this should alleviate some of the type issues.

**Comment by dmmc12** [comment]
Great effort Lucas!

**Pull request merged**